calico_rkt/calico_policy.py

```diff
@@ -0,0 +1,149 @@
+
+        :param endpoint:
+        :return: None
+        """
+        assert self.profile_name, "No profile name set."
+        if not self._client.profile_exists(self.profile_name):
+            # If the profile doesn't exist, create it.
+            _log.info("Creating new profile '%s'", self.profile_name)
+            rules = self.generate_rules()
+            self._client.create_profile(self.profile_name, rules)
+
+        # Set the default profile on this pod's Calico endpoint.
+        _log.info("Setting profile '%s' on endpoint %s",
+                  self.profile_name, endpoint.endpoint_id)
+        self._client.set_profiles_on_endpoint(profile_names=[self.profile_name],
+                                              endpoint_id=endpoint.endpoint_id)
+
+    def remove_profile(self):
+        """Remove the profile if there are no endpoints attached.
+
+        :return: None
+        """
+        raise NotImplementedError("Must implement method in subclass")
+
+    def generate_rules(self):
+        """Generates Calico Rules
+
+        :rtype: Calico Rules datatype or None
+        :return: rules - contains inbound and outbound policy rules; None
+        uses default rules
+        """
+        raise NotImplementedError("Must implement method in subclass")
+
+
+class DefaultPolicyDriver(BasePolicyDriver):
+    """
+    Implements default network policy for a generic CNI plugin.
+    """
+    def __init__(self, network_name):
+        BasePolicyDriver.__init__(self)
+        self.profile_name = network_name
+
+    def remove_profile(self):
+        """Right now this function is a no-op.
+
+        Need to think more about how to handle the race condition that exists
+        between removing a profile and creating a new one of the same name.
+
+        For now we'll leak profiles in Calico datastore.
+
+        :return: None
+        """
+        _log.info("Called remove profile. Doing nothing for now.")
+
+    def generate_rules(self):
+        """Generates rules for a default CNI plugin.
+
+        The default rules for a generic CNI plugin is to allow ingress traffic
+        from containers in the same network and allow all egress traffic.
+
+        :return: None - use default rules
+        """
+        return None
+
+
+class KubernetesDefaultPolicyDriver(BasePolicyDriver):
+    """
+    Implements default network policy for a Kubernetes container manager.
+    """
+    def __init__(self):
+        BasePolicyDriver.__init__(self)
+        self.profile_name = KUBERNETES_DEFAULT_PROFILE
+
+    def remove_profile(self):
+        """Right now this function is a no-op.
+
+        Need to think more about how to handle the race condition that exists
+        between removing a profile and creating a new one of the same name.
+
+        For now we'll leak profiles in Calico datastore.
+
+        :return: None
+        """
+        _log.info("Called remove profile. Doing nothing for now.")
```

Comment:
I'd reword this to be more clear.

"Not removing profile %s, clean up manually if desired"